src/shims/unix/fd.rs

```diff
@@ -380,6 +435,31 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         Ok((-1).into())
     }
 
+    /// Function used to update the ready list of an epoll_event. It takes in all the readiness
+    /// flags of a file description and return event as ready if any of the flag is monitored
+    /// by the epoll_event.
+    fn update_readiness(
+        &self,
+        ready_flags: u32,
+        epoll_events: &BTreeMap<(i32, i32), Weak<EpollEvent>>,
+    ) -> InterpResult<'tcx> {
+        for (_, event) in epoll_events.iter() {
+            if let Some(epoll_event) = event.upgrade() {
+                // Retrieve the same flag between file description readiness and epoll event and
+                // update the ready list.
```

Comment:
What does that mean, "retrieve the same flag between ... and ..."?

Reply:
Updated. This basically means if we have a file description that has readiness flag of (``EPOLLIN`` xor ``EPOLLOUT``), and we are interested in the ``EPOLLIN`` status of the file description, an ``EpollReturn`` that contains the ``EPOLLIN`` flag will be inserted to the ready list.